src/main/kotlin/com/ampnet/auditornode/controller/IpfsController.kt

```diff
@@ -1,26 +1,33 @@
 package com.ampnet.auditornode.controller
 
 import com.ampnet.auditornode.controller.documentation.IpfsControllerDocumentation
+import com.ampnet.auditornode.model.response.IpfsDirectoryUploadResponse
 import com.ampnet.auditornode.persistence.model.IpfsHash
+import com.ampnet.auditornode.persistence.model.NamedIpfsFile
 import com.ampnet.auditornode.persistence.repository.IpfsRepository
+import com.ampnet.auditornode.util.UuidProvider
 import io.micronaut.http.HttpResponse
 import io.micronaut.http.annotation.Controller
+import io.micronaut.http.multipart.CompletedFileUpload
 import mu.KotlinLogging
+import org.reactivestreams.Publisher
+import reactor.core.publisher.Flux
 import javax.inject.Inject
 
 private val logger = KotlinLogging.logger {}
 
 @Controller("/ipfs")
 class IpfsController @Inject constructor(
-    private val ipfsRepository: IpfsRepository
+    private val ipfsRepository: IpfsRepository,
+    private val uuidProvider: UuidProvider
```

Comment:
Is this needed. To create an interface an a class implementation just for `UUID.randomUUID()`?

Reply:
Yes, it is needed. This way there are no side-effects directly in code which cannot be removed in tests. By using a provider, I can use a mock value in test which returns an UUID which I want (and not always a random one). Same goes for things like current time - how would you test a method which has `now()` in it? Without a provider, it would be impossible to test it completely.